src/compiler/types/tests/ComponentCompilerMeta.stub.ts

```diff
@@ -0,0 +1,23 @@
+import * as d from '@stencil/core/declarations';
+
+/**
+ * Generates a stub {@link ComponentCompilerMeta}. This function uses sensible defaults for the initial stub. However,
+ * any field in the object may be overridden via the `overrides` argument.
+ * @param overrides a partial implementation of `ComponentCompilerMeta`. Any provided fields will override the
+ * defaults provided by this function.
+ * @returns the stubbed `ComponentCompilerMeta`
+ */
+export const stubComponentCompilerMeta = (
+  overrides: Partial<d.ComponentCompilerMeta> = {}
+): d.ComponentCompilerMeta => {
+  // TODO(STENCIL-378): Continue to build out default stub, remove the type assertion on `default`
```

Comment:
This is a pretty complex type - I decided to add what I need here, and we'll continue to build this out. STENCIL-378 has been in the backlog for a few weeks, so _technically_ I made progress there 😉